### PR TITLE
HomePageRoot ES6 conversion

### PR DESCRIPTION
**Pull request by jelliotartz**
Describe your changes.
- Converts HomePageRoot component from coffeescript to ES6.
- Adds tests for HomePageRoot component.
- Rather than passing in all props as was done in the coffeescript file, I looked at what props were being used in `HomePageForUser` and only passed in what was being used (`user` and `location`).
- Similarly, no props are used in `HomePageNotLoggedIn`, so I didn't pass any in.
- Question: is there an issue created for ES6 conversion? I searched Issues and couldn't find one to append this PR to.
- Staged [here](https://home-cs-to-js.pfe-preview.zooniverse.org/).

# Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch? 
- [x] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

## Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [x] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/pages/home.spec.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    beforeEach(function () {
+      wrapper = shallow(<HomePageRoot user={user} location={location} />)
+    });
+
+    it('without crashing', function () {
+      assert.equal(wrapper, wrapper)
+    });
+
+    it('with a user prop', function() {
```
Comment (srallen) [inline app/pages/home.spec.js]: This is testing the child component, not really `HomePageRoot`. This test would be more appropriate for testing `HomePageLoggedIn`.

File: app/pages/home.spec.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      const userDisplayName = wrapper.nodes[0].props.user.display_name
+      assert.equal(userDisplayName, 'Test User')
+    });
+
+    it('with a location prop', function() {
```
Comment (srallen) [inline app/pages/home.spec.js]: Same comment about testing the child, not `HomePageRoot`

File: app/pages/home.spec.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      wrapper = shallow(<HomePageRoot />)
+    })
+
+    it('without crashing', function () {
+      assert.equal(wrapper, wrapper)
+    });
+
+    it('for a not authenticated user', function() {
+      const child = wrapper.children().root.length
+      assert.equal(child, 1)
```
Comment (srallen) [inline app/pages/home.spec.js]: What you could do instead is `assert.equal(wrapper.find('HomePageNotLoggedIn').length, 1)`

File: app/pages/home.spec.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    })
+
+    it('without a user prop', function() {
```
Comment (srallen) [inline app/pages/home.spec.js]: This is testing the child component, not `HomePageRoot`

**Comment by jelliotartz** [comment]
Thanks for your comments on how to test just `HomePageRoot` and not its child components. I appreciate the feedback!

I'm not sure what why, but the syntax you're suggesting

```js
it('renders <HomePageNotLoggedIn /> when there isn\'t a user', function() {
  assert.equal(wrapper.find('HomePageNotLoggedIn').length, 1);
});
```

is not passing. It looks like it's trying to check the length of a ShallowWrapper object, and it's returning 0 instead of 1.

Adding `first()` or `root` eg

```js
it('renders <HomePageNotLoggedIn /> when there isn\'t a user', function() {
  assert.equal(wrapper.find('HomePageNotLoggedIn').root.length, 1)
});

it('renders <HomePageLoggedIn /> when there is a user', function() {
  wrapper.setProps({ user, location });
  assert.equal(wrapper.find('HomePageLoggedIn').first().length, 1)
});
```

creates passing tests, but I still think there's something wonky going on with the syntax or the assert library - I can interchange `HomePageLoggedIn` and `HomePageNotLoggedIn` in the assert statements, and both still pass, so it doesn't seem like it's doing what it's supposed to.

I thought maybe it has to do with not importing `HomePageLoggedIn` and `HomePageNotLoggedIn`, but I tried importing them directly, and did not find a difference in terms of tests passing and failing.

I've been looking at [collaborator.spec](https://github.com/zooniverse/Panoptes-Front-End/blob/9fc241547630148bd10a3fe97e5272b61ca8d6d7/app/collections/collaborators.spec.js) as an example of how to use the `assert` library's syntax. I'm at a loss to explain why `wrapper.find('Component').length` seems to be working differently in my spec than in collaborator spec.

File: app/pages/home.spec.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+const location = {
+  hash: '#page'
+};
+
+describe('HomePageRoot', () => {
+  let wrapper;
+
+  beforeEach(() => {
```
Comment (srallen) [inline app/pages/home.spec.js]: This can just be `before()`

**Comment by jelliotartz** [comment]
Gah! Thank you!! I'm tempted to put a 'Common Gotchas' PR up on the Enzyme docs to put that little detail out there!

**Pull request merged**